roles/lib_utils/library/oo_azure_rm_publish_image.py

```diff
@@ -160,6 +174,26 @@ def prepare_action(self, url, action='GET', data=None, add_headers=None):
 
         return AzurePublisher.request(action.upper(), self.server + url, data, headers)
 
+    def cancel_and_wait_for_operation(self, params):
+        '''cancel the current publish operation and wait for operation to complete'''
+
+        now = time.time()
+        # we need to wait here for 'submissionState' to move to 'canceled'
+        while True:
```

Comment:
shouldn't the self.cancel_operation should happen once before the loop?
is the timeout necessary?

Reply:
I could call cancel first or I could get the current `submissionState` and verify if it is `canceled` already.  I ran into a few race conditions where I could call publish/cancel and the get_operations would return that it was in state `running`.  If someone has previously canceled then this code doesn't require another cancel to be issued.

Thoughts?